### DISCUSSION TITLE
Future direction of BoxBilling

### DISCUSSION THREAD

**Discussion by j-a-pope** [boxbilling/boxbilling#933] [Ideas]
I think that the direction of this software is currently a little bit unclear, and I am curious what other people actually expect or want from it. 

At the moment it seems to not really know what it wants to be. 

I guess essentially the question is whether people are looking for a big piece of software that does everything, or a light framework that does all of the core functions really well, and is easily extensible with additional 'modules',  'plugins', or whatever you choose to call them. 

Personally I would like to see it move towards the second of those options.

For example:

- I don't think that file manager belongs in there - I think that is something for a server control panel, not for a billing and client management portal. 

- I also don't think that the forum should be installed by default, if it is there as part of the core at all then it should definitely be an option on install. It's far from the best forum solution out there and I think it would be better to have nice integration with something better if you want a forum at all. 
 
It would be good to ship with a limited number of payment integrations, control panel integrations, etc. and have  a clear well-maintained repository of others that can be added. 

What are your views on what you want from BB and what direction the codebase should go in?

**Comment by notbasetwo** [comment]
I agree with this. There are a lot of things that are unclear with this project going forward.

Your comments on the file manager & forum module are bang on target, BoxBilling was always a little confused with its direction back in the day and I think that both of these elements are a big sign of that.

I am relatively interested in the project - not really for my own use, but perhaps as something to spend a bit of time on. 

My main concern is around governance and the relationship the new maintainers have with the old ones, particularly when it comes to copyright over the logo, etc as they would not necessarily covered by the license. I do not see any trademark registered on the few databases I checked, so name is not a concern as far as I am aware. If it is a new team with most of the old maintainers jumping ship, why on earth is this not a fork?

Everything seems to be conducted on an invite only Slack if the documentation is to be believed.

**Comment by yagiz-dev** [comment]
@notbasetwo


https://join.slack.com/t/boxbilling/shared_invite/zt-u48e7wa8-eHnvZMu7c_JtmYCQd39bIQ

Not sure if this works, but give it a try.

**Comment by j-a-pope** [comment]
@notbasetwo I think that governance is a potential issue and all a bit confused at the moment. 

@fordnox is the current remaining maintainer of BoxBilling, and would be the best person to comment on it, I believe that he is still interested in the project but isn't currently very active here. 

My personal take on governance is that it needs to be much more transparent and clear. It is already much less confusing than it was a year ago, but there are some things that really have to be fixed urgently, like removing some of the wrong information that is still on the .com website, making it clear who maintainers are, and outlining the future direction and roadmap. It would also be good for there to be a clear process and understanding of how decisions are made instead of people randomly making commits, etc. that might contradict each other.

The ideal as far as I can see, if there is a real desire to make this a large and trusted project, would be for the ownership to transition to a non-profit foundation or cooperative. Also there is scope for donations to the project but instead of a random Paypal button this should be transparent using something like Open Collective.

**Reply by BelleNottelling** [reply]
> making it clear who maintainers are

Actually, at the moment this is kinda up to all of use who are members of the BoxBilling org. By default our membership is hidden to people who aren't part of it, but members can change it so their own membership is public

**Reply by notbasetwo** [reply]
This is all super important, yes. 

I would be very interested in hearing from the current maintainer about this, good shout!

As far as transferring to a non-profit, I think this would be the best end goal, as then there would be a stable platform for governance going forward. However in the intermediate, a statement of intent from the current copyright holder would probably suffice.

**Comment by notbasetwo** [comment]
As an aside, perhaps switching from Slack to Gitter would be a worthwhile endeavour. It is both FLOSS and much more appropriate for an open project.

**Reply by yagiz-dev** [reply]
Never used Gitter, but I'm okay with it as long as others agree to move.

**Reply by BelleNottelling** [reply]
I like gitter.  The app never gave me notifications though, so imo dev talk should be kept to slack and gitter can be used for the community

**Comment by xSido** [comment]
So... is there any live/fast chat method used ? (like discord)

**Reply by BelleNottelling** [reply]
We have the slack, but as far as I know it's invite only and I can't find a way to create an invite link

**Reply by xSido** [reply]
to invite people in slack, you have to swipe from left to right (on phone), then click on the 3 dots of the workspace u want to invite, and select Invite Members.

**Reply by yagiz-dev** [reply]
> to invite people in slack, you have to swipe from left to right (on phone), then click on the 3 dots of the workspace u want to invite, and select Invite Members.

He probably knows this. Only @fordnox is admin in Slack and AFAIK he's the only one who can create invite links.

**Reply by xSido** [reply]
i find discord easier to manage such project. is it much to ask to give it a try :D ?

**Reply by yagiz-dev** [reply]
> i find discord easier to manage such project. is it much to ask to give it a try :D ?

I personally agree, but everyone has different opinions on that.

As fordnox is currently the only admin in the project and he's been inactive since months, the administration part of the project seems to be struggling.

We have (had) a Discord server, but the only person who can arrange the server is also Andrius. I just hope he comes back in the near future, there's a lot of stuff we need to discuss with him.

**Reply by xSido** [reply]
I see commits more recent than when they went inactive.

**Reply by timothygwebb** [reply]
@fordnox When you have time please discuss this with the team and assign privileges as you feel appropriate. Thanks Timothy Webb Sr.

**Reply by xSido** [reply]
Maybe i can help.
Things i might be useful, discord\forum\community manager\moderator, website\front end, hosting, graphic design.

**Reply by BelleNottelling** [reply]
@xSido does this link work for you? This is one I was given beforehand. 
https://join.slack.com/t/boxbilling/shared_invite/zt-u48e7wa8-eHnvZMu7c_JtmYCQd39bIQ
Also, I'm aware of how to create the invite, however, I can only enter an email and REQUEST that an invite is created for that email address.

**Reply by xSido** [reply]
> @xSido does this link work for you? This is one I was given beforehand.
> https://join.slack.com/t/boxbilling/shared_invite/zt-u48e7wa8-eHnvZMu7c_JtmYCQd39bIQ
> Also, I'm aware of how to create the invite, however, I can only enter an email and REQUEST that an invite is created for that email address.

it does not work for me.

**Reply by notbasetwo** [reply]
As far as Discord goes, that is far too much merging of my non-work/personal stuff with other things. Slack can be disabled entirely when I don't have time to deal with things.

**Comment by ShadowCrownServices** [comment]
Yes, definitely keep it a lightweight framework with the option to install other functions as and when needed.

Personally, most site owners/resellers usually build their own forums... so perhaps focusing the effort on tickets (direct customer service) would be more of a prominent thought.

Creat better/wider integration, such as an option who to connect with (Server), Solus, Plesk, Etc

Along with having a little more description/depth within help files; also, making your Extensions page work would be helpful too.
